src/main/java/core/basesyntax/dao/ReportDao.java

```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.Report;
+
+public interface ReportDao {
+    Report save(Report report);
```

Comment:
Do you really want to make it that difficult? You don't have to make the report object, this method can be simplified. You don't need dao to handle the reference. Dao works with the database and is the data carrier from the DB to our services. You are violating the DIP principle here. Report should be a service, not a high-level module. How about creating a service that has two methods of saving to your DB and making a count that will return a string. You can pass this string to your FileWriter along with the path and create a reference file.

Reply:
I made Report, Fruit, and Activity as Objects because in java we are working with objects and they are the main entities (models) of our program. And when we using a database we don't parse their simple value but the whole object the same we get from DB the whole object. I tried to make it as abstract as possible and abstract from the file as the source and target of information. For example what you will do when you will have to save your report to database not to the file?

Reply:
The complexity of the architecture is determined by the complexity of the task. Your application works, but is it worth it? When we know that we're writing the readout to a file and not sending it to the DB. Neither you nor I know what to do if we need to load data into the DB instead of the file, let's be honest. If you're already implementing so much detail and thinking through the architecture, then let's start with the fact that your abstract interface is implemented wrong and looks canonically wrong. I'm not even talking about CRUD methods right now. I still insist that you redo your solution, but it's up to you.

Reply:
I know a little bit.

Reply:
I have some experience working with spring and hibernate. I don't insist that it is 100% correct but I would prefer to live it as it is right now for mentors

Reply:
I agree, I'm not sure I'm right either.

Reply:
Approved